packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -417,13 +429,27 @@ - (void)setupEventSinkIfReadyToPlay {
     AVAsset *asset = currentItem.asset;
     if ([asset statusOfValueForKey:@"tracks" error:nil] != AVKeyValueStatusLoaded) {
       void (^trackCompletionHandler)(void) = ^{
-        if ([asset statusOfValueForKey:@"tracks" error:nil] != AVKeyValueStatusLoaded) {
+        NSError *error;
+        switch ([asset statusOfValueForKey:@"tracks" error:&error]) {
+          case AVKeyValueStatusLoaded:
+            // This completion block will run on an AVFoundation background queue.
+            // Hop back to the main thread to set up event sink.
+            [self performSelector:_cmd
+                         onThread:NSThread.mainThread
+                       withObject:self
+                    waitUntilDone:NO];
+            break;
           // Cancelled, or something failed.
-          return;
+          case AVKeyValueStatusFailed: {
+            // if something failed then future should result in error
+            dispatch_async(dispatch_get_main_queue(), ^{
+              [self sendFailedToLoadVideoEvent:error];
```

Comment:
is this code the key change that fixed the bug?

Reply:
Seems no, I think it is that change in `onListenWithArguments`. But without that change it seems also this change works as I now tested it with "Code sample 1" from https://github.com/flutter/flutter/issues/151475.